### PR TITLE
[On Hold] Proposal to add generic pluginInputs field in backup/restore/schedule CRs

### PR DESCRIPTION
**Pull request by anshulahuja98**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: design/plugininputs-field.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#### Use case 1: Change StorageClass RIA (velero.io/change-storage-class)
+Change Storage Class plugin, uses a global configmap which is discovered through labels/ annotations which is not ideal customer experience and prohibhit usage of specific settings for each backup/ schedule setup.
+
+The pluginInputs field can be used to pass in the reference the configmap/ secret to use for a particular plugin. This will allow the user to use different configmaps/ secrets for different backups/ schedules.
+
+Example: 
+```yaml
+spec:
+    pluginInputs:
+    - name: velero.io/storageclass
```
Comment (blackpiglet) [inline design/plugininputs-field.md:45]: Is the `name` field used to represent the plugin's name?
If so, the name should be `velero.io/change-storage-class`.

File: design/plugininputs-field.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+Example: 
+```yaml
+apiVersion: velero.io/v1
+kind: Backup
+metadata:
+  name: backup-1
+spec:
+    pluginInputs:
+    - name: velero.io/csi
```
Comment (blackpiglet) [inline design/plugininputs-field.md:75]: The plugin name should be `velero.io/csi-volumesnapshot-backupper`.
I also wonder if the configuration is applied to many plugins, and we can only choose the plugin by name, then this gonna be a long list.
Could we consider adding some RE function here?

File: design/plugininputs-field.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    - properties:
+        - key: csi.cloud.disk.driver
```
Comment (blackpiglet) [inline design/plugininputs-field.md:77]: I think the format should be 
```yaml
    - properties:
        - csi.cloud.disk.driver: csi-diskdriver-snapclass
        - csi.cloud.file.driver: csi-filedriver-snapclass
```

**Comment by anshulahuja98** [comment]
@blackpiglet thanks for going through this PR and providing feedback, currently this PR is on hold since we could not get agreement from community for implementing this.

I'll address your comments in case in future we pick it again.

**Pull request closed without merging**